### PR TITLE
add workflow side bar for viewer mode

### PR DESCRIPTION
**Pull request by MauricioUyaguari**
## Summary

Closes https://github.com/finos/legend-studio/issues/779

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

viewer_mode_1
.com/finos/legend-studio/tree/master/docs/ux

viewer_mode_3

### CODE REVIEW

File: .changeset/gold-chicken-return.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@finos/legend-studio": patch
```
Comment (akphi) [inline .changeset/gold-chicken-return.md:2]: --> `minor`

File: packages/legend-studio/src/stores/sidebar-state/WorkflowsState.ts [hunk 2]
```diff
@@ -437,3 +459,70 @@ export class WorkspaceWorkflowsState {
...
+  get label(): string {
+    return `PROJECT WORKFLOWS ${this.version ? `VERSION ${this.version}` : ''}`;
+  }
+
+  override get version(): string | undefined {
+    return this._version?.id.id;
+  }
+}
+
+export class UnsupportedWorkflowsState extends WorkflowsState {
```
Comment (akphi) [inline packages/legend-studio/src/stores/sidebar-state/WorkflowsState.ts:513]: This is not a bad idea, but why don't we just do `workflowState?: WorkflowState`. This seems quite overkill to me...

**Review by akphi — CHANGES_REQUESTED** [review]
Minor comments. Thanks for doing it this way. I think you start to click with my `abstract class everything` vibe 👨‍🎤

File: packages/legend-studio/src/components/editor/side-bar/WorkflowManager.tsx [hunk 3]
```diff
@@ -420,111 +418,159 @@ const WorkflowTreeNodeContainer: React.FC<
...
-      <div className="panel__header side-bar__header">
-        <div className="panel__header__title workspace-workflows__header__title">
-          <div className="panel__header__title__content side-bar__header__title__content">
-            WORKSPACE WORKFLOWS
+    return (
+      <div className="panel workspace-workflows">
+        <div className="panel__header side-bar__header">
+          <div className="panel__header__title workspace-workflows__header__title">
+            <div className="panel__header__title__content side-bar__header__title__content">
+              {workflowManagerState.label}
```
Comment (akphi) [inline packages/legend-studio/src/components/editor/side-bar/WorkflowManager.tsx:490]: Shou;dn't we call this `Workflow Manager`?

File: packages/legend-studio/src/components/editor/side-bar/WorkflowManager.tsx [hunk 4]
```diff
@@ -420,111 +418,159 @@ const WorkflowTreeNodeContainer: React.FC<
...
-              data-testid={
-                LEGEND_STUDIO_TEST_ID.SIDEBAR_PANEL_HEADER__CHANGES_COUNT
-              }
-            >
-              {workflowsState.workflowStates.length}
-            </div>
+    );
+  },
+);
+export const UnsupportedWorkflowManager = observer(() => (
```
Comment (akphi) [inline packages/legend-studio/src/components/editor/side-bar/WorkflowManager.tsx:534]: Let's remove this

File: packages/legend-studio/src/components/viewer/Viewer.tsx [hunk 5]
```diff
@@ -157,6 +163,34 @@ const ViewerStatusBar = observer(() => {
...
+const ViewerSideBar = observer(() => {
+  const viewerStore = useViewerStore();
+  const editorStore = viewerStore.editorStore;
+  const renderSideBar = (): React.ReactNode => {
+    switch (editorStore.activeActivity) {
+      case ACTIVITY_MODE.EXPLORER:
+        return <Explorer />;
+      case ACTIVITY_MODE.PROJECT_OVERVIEW:
+        return <ProjectOverview />;
+      case ACTIVITY_MODE.WORKFLOWS:
```
Comment (akphi) [inline packages/legend-studio/src/components/viewer/Viewer.tsx:175]: Let's not even render this when there's no workflow state

File: packages/legend-studio/src/stores/ViewerStore.ts [hunk 6]
```diff
@@ -509,4 +516,17 @@ export class ViewerStore {
       onLeave(false);
     }
   }
+
+  initWorkflowsState(): void {
```
Comment (akphi) [inline packages/legend-studio/src/stores/ViewerStore.ts:520]: --> `initWorkflowManagerState`

File: packages/legend-studio/src/stores/sidebar-state/WorkflowManagerState.ts [hunk 7]
```diff
@@ -437,3 +521,53 @@ export class WorkspaceWorkflowsState {
...
+      fetchAllWorkflows: flow,
+    });
+    this._version = version;
+  }
+
+  get label(): string {
+    return `PROJECT WORKFLOWS ${this.version ? `VERSION ${this.version}` : ''}`;
+  }
+
+  override get version(): string | undefined {
```
Comment (akphi) [inline packages/legend-studio/src/stores/sidebar-state/WorkflowManagerState.ts:570]: Let's call `_version` `version` and this as `versionId`?

File: packages/legend-studio/src/stores/sidebar-state/WorkflowManagerState.ts [hunk 8]
```diff
@@ -437,3 +521,53 @@ export class WorkspaceWorkflowsState {
...
+      fetchWorkflowsState: observable,
+      logState: observable,
+      workflowStates: observable,
+      fetchAllWorkflows: flow,
+    });
+    this._version = version;
+  }
+
+  get label(): string {
+    return `PROJECT WORKFLOWS ${this.version ? `VERSION ${this.version}` : ''}`;
```
Comment (akphi) [inline packages/legend-studio/src/stores/sidebar-state/WorkflowManagerState.ts:567]: Maybe I mistaken, but I thought this is for the header title of the sidebar panel, then we should always keep it as `Workflow Manager`, yeah? If so we can remove this `label()`

File: packages/legend-studio/src/stores/EditorConfig.ts [hunk 9]
```diff
@@ -60,7 +60,7 @@ export enum ACTIVITY_MODE {
   SETTINGS = 'SETTINGS',
   REVIEW = 'REVIEW',
   PROJECT_OVERVIEW = 'PROJECT_OVERVIEW',
-  WORKSPACE_WORKFLOWS = 'WORKSPACE_WORKFLOWS',
+  WORKFLOWS = 'WORKFLOWS',
```
Comment (akphi) [inline packages/legend-studio/src/stores/EditorConfig.ts:63]: --> `WORKFLOW_MANAGER`

File: packages/legend-studio/src/stores/ViewerStore.ts (hunk 6 above)
Comment (akphi) [inline packages/legend-studio/src/stores/ViewerStore.ts:520]: --> `initWorkflowManagerState`

**Review by akphi — CHANGES_REQUESTED** [review]

File: packages/legend-studio/src/components/editor/side-bar/WorkflowManager.tsx [hunk 10]
```diff
@@ -256,8 +256,8 @@ const WorkflowJobLogsViewer = observer(
 const WorkflowExplorerContextMenu = observer(
   (
     props: {
-      workflowsState: WorkspaceWorkflowsState;
-      workflowState: WorkspaceWorkflowState;
+      workflowsState: WorkflowManagerState;
```
Comment (akphi) [inline packages/legend-studio/src/components/editor/side-bar/WorkflowManager.tsx:259]: --> `workflowManagerState`
Let's rename all `workflowsState` to `workflowMangerState`

File: packages/legend-studio/src/stores/sidebar-state/WorkflowManagerState.ts [hunk 11]
```diff
@@ -91,14 +103,14 @@ const updateWorkflowJobData = (
...
-  sdlcState: EditorSDLCState;
+  workflowsState: WorkflowManagerState;
   job: WorkflowJob | undefined;
   logs: string;
   fetchJobLogState = ActionState.create();
 
   constructor(
     editorStore: EditorStore,
-    sdlcState: EditorSDLCState,
+    workflowsState: WorkflowManagerState,
```
Comment (akphi) [inline packages/legend-studio/src/stores/sidebar-state/WorkflowManagerState.ts:113]: --> `workflowManagerState`

File: packages/legend-studio/src/stores/sidebar-state/WorkflowManagerState.ts [hunk 12]
```diff
@@ -226,21 +256,30 @@ export class WorkspaceWorkflowState {
     return treeData;
   }
 
-  *fetchAllWorkspaceWorkJobs(
+  *fetchAllWorkJobs(
```
Comment (akphi) [inline packages/legend-studio/src/stores/sidebar-state/WorkflowManagerState.ts:259]: -> `fetchAllWorkflowJobs`

**Review by akphi — CHANGES_REQUESTED** [review]

File: packages/legend-studio/src/stores/sidebar-state/WorkflowManagerState.ts [hunk 13]
```diff
@@ -437,3 +404,279 @@ export class WorkspaceWorkflowsState {
     }
   }
 }
+
+export class WorkspaceWorkflowManagerState extends WorkflowManagerState {
+  constructor(editorStore: EditorStore, sdlcState: EditorSDLCState) {
+    super(editorStore, sdlcState);
+    makeObservable(this, {
```
[the remark this answers is not in the document]
Comment (MauricioUyaguari) [reply packages/legend-studio/src/stores/sidebar-state/WorkflowManagerState.ts:411]: Nope its not.

[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-studio/src/stores/sidebar-state/WorkflowManagerState.ts]: @MauricioUyaguari  image

**Review by akphi — APPROVED** [review]

**Pull request merged**